### PR TITLE
CP-37241: Disable machine account on domain leave

### PR DESCRIPTION
**Pull request by liulinC**
PBIS just disable machine account during domain leave
This commit let winbind keep this behavior for backward compatibility
instead of delete the machine account

Signed-off-by: Lin Liu <lin.liu@citrix.com>

### CODE REVIEW

File: ocaml/xapi/extauth_plugin_ADwinbind.ml [hunk 1]
```diff
@@ -484,17 +484,17 @@ let persist_extauth_config ~domain ~user ~ou_conf ~workgroup =
...
   |> Server_helpers.exec_with_new_task "update external_auth_configuration"
 
-let clean_machine_account ~service_name = function
+let disable_machine_account ~service_name = function
   | Some u, Some p -> (
-      (* Clean machine account in DC *)
+      (* Disable machine account in DC *)
       let env = [|Printf.sprintf "PASSWD=%s" p|] in
-      let args = ["ads"; "leave"; "-U"; u; "-d"; debug_level] in
+      let args = ["ads"; "leave"; "-U"; u; "--keep-account"; "-d"; debug_level] in
```
Comment (lippirk) [inline ocaml/xapi/extauth_plugin_ADwinbind.ml:491]: When you leave an AD domain using XenCenter, you can choose whether to leave the machine account enabled or not, so I would have expected to see some boolean switch alongside `--keep-account`?

Comment (liulinC) [reply ocaml/xapi/extauth_plugin_ADwinbind.ml:491]: if user name and password is not provides (in XenCenter), then the machine account is not touched, just local leave is performed. 
if user name and password is provided (in XenCenter), we originally  delete the machine account, but we now just disable the machine account like PBIS

**Review by lippirk — APPROVED** [review]

**Comment by liulinC** [comment]
@lippirk @lindig @psafont  Thanks for the help, 
I added to new commit, could you please help to review again?

File: ocaml/tests/test_data/repository_pkg_of_fullname_all [hunk 2]
```diff
@@ -253,7 +253,6 @@ intel-igb-5.3.5.20-2.xs8.x86_64
 nss-pem-1.0.3-4.el7.x86_64
 gssproxy-0.7.0-17.el7.x86_64
 nss-sysinit-3.36.0-7.el7_5.x86_64
-pbis-open-8.2.3-1.7.8.xs8.x86_64
```
Comment (lippirk) [inline ocaml/tests/test_data/repository_pkg_of_fullname_all:256]: why have these been removed?

**Comment by lippirk** [comment]
Looks fine to me, once the `ocaml/tests/test_data/repository_pkg_of_fullname_all` changes have been reverted

Another note is that the 'Avoid using List.hd' commit contains some unrelated changes

**Review by lippirk — APPROVED** [review]

File: ocaml/tests/test_data/repository_pkg_of_fullname_all (hunk 2 above)
Comment (liulinC) [reply ocaml/tests/test_data/repository_pkg_of_fullname_all:256]: We are going to remove PBIS related packages, 
I know this is just for unittest, but remove it makes no bad.

**Review by liulinC — COMMENTED** [review]
> Looks fine to me, once the `ocaml/tests/test_data/repository_pkg_of_fullname_all` changes have been reverted
> 
> Another note is that the 'Avoid using List.hd' commit contains some unrelated changes

Yes, Just merged them into this commit, to avoid too many commits.
Others just some minor refine.

**Review by robhoes — APPROVED** [review]

**Pull request merged**